pkg/plugins/manager/loader/loader.go

```diff
@@ -195,6 +203,16 @@ func (l *Loader) loadPlugins(ctx context.Context, class plugins.Class, pluginJSO
 			return nil, err
 		}
 		metrics.SetPluginBuildInformation(p.ID, string(p.Type), p.Info.Version, string(p.Signature))
+
+		if l.features.IsEnabled(featuremgmt.FlagAccessControlOnCall) && len(p.JSONData.Roles) > 0 {
```

Comment:
technically, the feature toggle being checked in `DeclarePluginRoles` as well, this is a bit overkill.